Integration 1

hkrishna opened the pull request:
Has the following branches:
1. dryer-tilestache #1 
2. better-error-responses #5 
3. add-opensciencemap-vectiles #4 
4. add-mapbox-vectiles #7 
5. add-layer-name-to-vtm-tiles #8 
6. drop-null-val-prop #9 
7. extend-multiprovider #3
8. update-column-names #12  
9. empty-mapbox-responses #13 
10. empty-oscimap-responses #14 
11. performance-updates #16 
12. vtm-update-heights #17 
13. remove-sort #18 
14. mapbox-content-type #21 
15. vtm-add-geometry-type #27

bcamper commented:
@hkrishna, this branch appears to be missing at least the following commits, which are needed to avoid seams between tiles in OSciMap tiles:

https://github.com/mapzen/TileStache/commit/35123fd3d9e11cc831d2eacf7edbfde27ca29122
https://github.com/mapzen/TileStache/commit/2186f358fa2ba82cf0634960ca62052a4ea95bab

zerebubuth committed:
Better documentation of the dependency between `road_sort_key` and `parse_layer_as_float`.

zerebubuth committed:
Merge pull request #65 from mapzen/194-layer-attribute

Add filter for parsing layer.

zerebubuth committed:
Set kind to `aboriginal_lands` for first nations state boundaries.

zerebubuth committed:
Merge pull request #66 from mapzen/284-first-nations-states

Set kind to `aboriginal_lands` for first nations state boundaries.

rmarianski committed:
Only include labels with name or sport tags

rmarianski committed:
Use representative_point instead of centroid

rmarianski committed:
Merge pull request #67 from mapzen/update-dynamic-label-behavior

Update label feature generation behavior

rmarianski committed:
Ensure original features always get added

Previously, if a feature wasn't a candidate for adding a label, it did
not get added to the layer. This corrects that.

rmarianski committed:
Merge pull request #68 from mapzen/restore-missing-features

Ensure original features always get added

zerebubuth committed:
Added an implementation of orient which can handle multi-polygons and other geometry types without throwing an exception.

zerebubuth committed:
Merge pull request #69 from mapzen/fix-orient-multi-polygon

Fix orientation for multi-polygons

rmarianski committed:
Version bump -> 0.4.0

nvkelso commented:
This high level view of the delta between mainline TileStache and Mapzen's fork is still interesting, but is out of date.

zerebubuth committed:
Added a post-process function to generate address points from building outlines. Updated building logic to remove 'POI-like' kinds - these will go in the POIs layer instead.

zerebubuth committed:
Make admin boundaries post-processing filter work with boundary linestring fragments rather than needing an oriented polygon.

zerebubuth committed:
Merge pull request #71 from mapzen/302-reduce-boundary-payload-size

Make admin boundary post-processor work with lines.

zerebubuth committed:
Add zoom level filter to address point generator.

zerebubuth committed:
Suppress building names when they are copies of the address, or convert them to addresses if they are numeric.

zerebubuth committed:
Fix logic for address points already in the data.

rmarianski committed:
Version bump -> 0.4.1

zerebubuth committed:
As per comment, we do want house numbers on the label points, so leave that property set.

zerebubuth committed:
Don't remove properties from the polygon any more.

zerebubuth committed:
Added post-process filter to drop certain features. This is needed so that we can generate landuse labels without having landuse polygons.

zerebubuth committed:
Fix issue with footpaths not showing around Civic Center Plaza.

zerebubuth committed:
Add function to drop the area property if it's zero, as this isn't useful information (the area probably isn't zero, just unspecified).

zerebubuth committed:
Extract multi-geometry type constructor logic to higher level.

zerebubuth committed:
Use more descriptive exception type.

zerebubuth committed:
Move shape type filter up to avoid doing more work than we have to.

zerebubuth committed:
Treat str, unicode the same way.

zerebubuth committed:
Made docstring more clear about values for the property. Updated property getting/dropping logic to be clearer. Dropped non-idiomatic use of pass in favour of continue.

zerebubuth committed:
Merge pull request #72 from mapzen/201-sort-out-landuse-buildings-pois

Sort out generated features for landuse, buildings & POIs

zerebubuth committed:
Fixed typo - method is called pop, not drop.

zerebubuth committed:
Merge pull request #73 from mapzen/201-sort-out-landuse-buildings-pois

Fixed typo - method is called pop, not drop.

zerebubuth committed:
Added post-process filter to remove duplicate features.

Where being a duplicate is defined by having some subset of
properties in common and being within a certain distance of the
"original" feature.

Feature order is not changed, so more features should be sorted
into importance order by the sorting step which runs at the end of
the filter functions.

The type of feature to filter out may be selected using the
geometry filter. This is primarily of use when filtering out
duplicate POIs / labels only.

zerebubuth committed:
Merge pull request #74 from mapzen/267-filter-duplicate-pois

Added post-process filter to remove duplicate features.

zerebubuth committed:
Sort by number of subway lines descending.

zerebubuth committed:
Move to_float to its own file, as we need to re-sort after merging subway stations, which means importing the POIs sort function and we can't do circular imports.

zerebubuth committed:
Add post-process filter to normalise and merge station POIs and another to keep only N features of a certain type in a layer.

zerebubuth committed:
Add function to rank POIs of a certain type within the layer, which can be useful for the client.

zerebubuth committed:
Made logic more readable, more efficient and better documented.

zerebubuth committed:
Merge pull request #75 from mapzen/268-too-many-subway-stations

Prioritise subway station features.

zerebubuth committed:
Snap geometry coordinates to a static grid. This helps paper over numerical rounding differences between different import pipelines.

zerebubuth committed:
Dirty hack to treat shapes with _exactly_ the same area as identical.

zerebubuth committed:
Add extra check to avoid duplicate boundary geometries.

zerebubuth committed:
Use ValueError instead of plain Exception.

zerebubuth committed:
Merge pull request #76 from mapzen/303-fix-water-boundaries

Fix water boundaries

zerebubuth committed:
Update sort order to account for service values.

zerebubuth committed:
Merge pull request #78 from mapzen/332-update-road-zooms

Update sort order to account for service values.

zerebubuth committed:
Try using the 'buffer' trick to make polygons valid after snapping. If that doesn't work, drop them and continue. Better to have a tile missing some water boundaries than no tile at all.

zerebubuth committed:
Synthesize a volume attribute for the client to filter on.

zerebubuth committed:
Synthesize volume as a separate function.

zerebubuth committed:
Add post-process filter function to ensure certain numeric minima. Used to double-check the filter on building area / volume before sending data to the client.

zerebubuth committed:
Merge pull request #79 from mapzen/77-fix-polygons-after-snapping

Try using the 'buffer' trick to make polygons valid after snapping

zerebubuth committed:
Process aerialways and try to keep values distinct from roads.

zerebubuth committed:
Merge pull request #81 from mapzen/212-filter-by-building-volume

Synthesize a volume attribute for the client to filter on.

zerebubuth committed:
Merge remote-tracking branch 'origin/integration-1' into 189-add-aerialways

zerebubuth committed:
Merge pull request #82 from mapzen/189-add-aerialways

Process aerialways and try to keep values distinct from roads.

zerebubuth committed:
Filter out features with empty subway_lines, as this is misleading.

zerebubuth committed:
Rename: subway lines -> transit routes.

zerebubuth committed:
Merge pull request #83 from mapzen/268-subway-route-updates

Updates for transit routes

zerebubuth committed:
Add support for dropping certain keys when copying them for label features.

zerebubuth committed:
Merge pull request #84 from mapzen/333-drop-internal-property

Add support for dropping certain keys on labels

zerebubuth committed:
Extracted deduplicator to a class, makes logic easier to read. Allowed the duplicate remover to work across several layers. Requires a bit of abnormal fiddling with the layers to get it to work.

zerebubuth committed:
Implement copy post-processor, which copies matching features to another layer.

zerebubuth committed:
Merge pull request #85 from mapzen/266-density-filter-pois

Made duplicate remove filter work across multiple layers

zerebubuth committed:
Volume should be an int.

zerebubuth committed:
Added post-processor function to replace geometry with representative point. This is for use in the POIs layer, where previously we were using a PostGIS-calculated centroid.

zerebubuth committed:
Merge pull request #87 from mapzen/212-volume-should-be-integer

Volume should be an int.

zerebubuth committed:
Merge pull request #88 from mapzen/fix-poi-centroid-reprensentative-point

Use reprensentative point for POIs, not centroid.

zerebubuth committed:
Replaced subway lines with transit routes, but needed to update the sort function too.

zerebubuth committed:
Merge pull request #89 from mapzen/fix-sort-transit-routes

Replaced subway lines with transit routes, but needed to update the s…

rmarianski committed:
Remove neighbourhood scalerank

rmarianski committed:
Update places sort to include n_photos and area

rmarianski committed:
Update _match_props to support value lists

rmarianski committed:
Merge pull request #91 from mapzen/wof-neighbourhoods-transform-sort-updates

Update transform and sort in places for WOF

zerebubuth committed:
Try not to split up roads, or merge them back together if possible.

zerebubuth committed:
Merge pull request #92 from mapzen/384-fix-broken-up-roads-on-boundary

Try not to split up roads, or merge them back together if possible.

rmarianski committed:
Add transform to drop feature properties

rmarianski committed:
Update n_photos property name to mz_n_photos

rmarianski committed:
Merge pull request #93 from mapzen/add-drop-properties-transform

Add transform to drop feature properties

rmarianski committed:
Merge pull request #94 from mapzen/update-n_photos-property-name

Update n_photos property name to mz_n_photos

rmarianski committed:
Add sort by `min_zoom` to `places` layer

rmarianski committed:
Merge pull request #95 from mapzen/add-sort-min_zoom-places

Add sort by `min_zoom` to `places` layer

rmarianski committed:
Version bump -> 0.5.0

rmarianski committed:
Protect against empty snap geometries

rmarianski committed:
Merge pull request #96 from mapzen/protect-from-empty-snap

Protect against empty snapped geometries

rmarianski committed:
Version bump -> v0.5.1

nvkelso committed:
Update CHANGELOG.md

rmarianski committed:
Add transform to make population an integer

rmarianski committed:
Remove population from props if parsing fails

rmarianski committed:
Merge pull request #97 from mapzen/population-integer

Add transform to make population an integer

zerebubuth committed:
Added piste as a kind of road and function to remove abandoned pistes.

zerebubuth committed:
Add default for pop.

zerebubuth committed:
Merge pull request #98 from mapzen/342-add-piste-routes

Add functions for piste routes

zerebubuth committed:
Give exit nodes the kind 'exit', as that's a bit easier to understand than a point with kind 'minor_road'.

zerebubuth committed:
Merge pull request #100 from mapzen/160-add-highway-exits

Proper kind for exit nodes.

zerebubuth committed:
Add racetracks as a kind of roads.

zerebubuth committed:
Merge pull request #102 from mapzen/344-add-winter-sports-pois

Add racetracks as a kind of roads.

zerebubuth committed:
Added function to grab the IATA code, if any, from the tags and put it on the feature.

zerebubuth committed:
Merge pull request #103 from mapzen/398-add-iata-codes

Add IATA code processing function.

zerebubuth committed:
Return `kind:path` for piers.

zerebubuth committed:
Merge pull request #104 from mapzen/382-add-pier-lines-to-roads

Return `kind:path` for piers.

zerebubuth committed:
Added explicit beach and winter sports landuse area sort orders.

zerebubuth committed:
Fixed bug: we default aerialway to the blank string, so it will always be `not None`. Further, testing `in` a single tuple doesn't make much sense, and that isn't how it's spelled anyway (should be `in ('chair_lift',)`.

zerebubuth committed:
Keep aerialways above other linear features.

zerebubuth committed:
Don't encroach on the reserved 'layer' range of orders.

zerebubuth committed:
Merge pull request #105 from mapzen/343-366-landuse-sort-order

Added explicit beach and winter sports landuse area sort orders.

zerebubuth committed:
Merge pull request #106 from mapzen/342-444-winter-sports-lines-sort-order

Sort order fixes for winter sports lines

zerebubuth committed:
Make winter sports areas show underneath forests/parks.

zerebubuth committed:
Merge pull request #107 from mapzen/343-reorder-winter-sports

Make winter sports areas show underneath forests/parks.

rmarianski committed:
Only include affirmative calculated road props

rmarianski committed:
Merge pull request #108 from mapzen/include-roads-properties-conditionally

Only include affirmative calculated road props

zerebubuth committed:
Updated changelog for v0.6.0.

rmarianski committed:
Add transform to include aeroway tag for kind=gate

zerebubuth committed:
Implement merging for linear features.

This also includes an ad-hoc implementation of feature dropping for
roads, which should be moved to a separate function and made
configurable for use in `queries.yaml`.

zerebubuth committed:
Add code to normalise leisure kinds for fitness-related POIs.

zerebubuth committed:
Merge pull request #110 from mapzen/465-gym-and-fitness-pois

Add code to normalise leisure kinds for fitness-related POIs.

zerebubuth committed:
Merge pull request #109 from mapzen/include-aeroway-for-kind-gate

Add transform to include aeroway tag for kind=gate

zerebubuth committed:
Merge remote-tracking branch 'origin/integration-1' into 358-merge-road-segments

zerebubuth committed:
Rewrite for clarity and move the drop-features functionality to config file.

zerebubuth committed:
Merge pull request #111 from mapzen/358-merge-road-segments

Merge road segments

zerebubuth committed:
Add function to normalise tourism kind and related properties.

zerebubuth committed:
Merge pull request #112 from mapzen/440-zoos-and-other-attractions

Add function to normalise tourism kind and related properties.

zerebubuth committed:
Update changelog for v0.7 release.

rmarianski committed:
Allow code in drop_features_where function

rmarianski committed:
Update docstring

rmarianski committed:
Merge pull request #113 from mapzen/update-drop_features_where

Allow code in drop_features_where function

zerebubuth committed:
Add bounding box clipping to exterior boundaries transform.

zerebubuth committed:
Merge pull request #114 from mapzen/502-water-boundaries-slow

Add bounding box clipping to exterior boundaries transform.

zerebubuth committed:
Add normalisation functions for social facilities and medical places.

zerebubuth committed:
Fix comment indent.

zerebubuth committed:
Merge pull request #115 from mapzen/526-inclusive-pois

Add normalisation functions for social facilities and medical places.

nvkelso committed:
fix #364 so all landuse has friendly sort_key orders

nvkelso committed:
set default landuse order to 11, and adjust everything else by +1

nvkelso committed:
things above the water should be uniquely above the water

nvkelso committed:
Merge pull request #116 from mapzen/nvkelso/landuse-sort

all landuse should have friendly sort_key orders

rmarianski committed:
Update road sort key values

rmarianski committed:
First round of updates

rmarianski committed:
Remove duplicate unclassified check

rmarianski committed:
Updates round 2

zerebubuth committed:
Use context object for parameters.

zerebubuth committed:
Add CSV property matching function.

zerebubuth committed:
Remove deprecated landuse sort key function.

zerebubuth committed:
Use cache to avoid re-reading the CSV file.

zerebubuth committed:
Use new 'resources' section for creating and storing the 'matcher' object. Also, matcher is now a callable.

zerebubuth committed:
Merge pull request #119 from mapzen/70-post-process-context

Use context object for parameters.

rmarianski committed:
Merge pull request #117 from mapzen/update-road-sort_key

Update road sort key values

zerebubuth committed:
Fix typo 'guage' -> 'gauge'.

zerebubuth committed:
Need layer for sort key - will have to delete it as a post-process step.

zerebubuth committed:
Added functionality to allow the column headings to specify what type they are.

zerebubuth committed:
Merge pull request #120 from mapzen/sort-key-csv-prototype

Sort key CSV prototype

rmarianski committed:
Support matching on zoom levels

rmarianski committed:
Add tag value, not key

rmarianski committed:
Merge pull request #121 from mapzen/support-zoom-matching

Support matching on zoom levels

rmarianski committed:
Add more precision to json formatter at z16

zerebubuth committed:
Fix typo - needs to use the key, rather than the string 'key'.

rmarianski committed:
Merge pull request #123 from mapzen/json-precision-z16

Add more precision to json formatter at z16

zerebubuth committed:
Merge pull request #124 from mapzen/fix-csv-matching-typo

Fix typo - needs to use the key, rather than the string 'key'.

rmarianski committed:
Add changelog for v0.8.0

nvkelso committed:
update changelog

nvkelso committed:
Merge pull request #125 from mapzen/changelog-v0.8.0

Add changelog for v0.8.0

zerebubuth committed:
After merging, simplify with a very small tolerance to remove duplicate and almost-colinear points.

zerebubuth committed:
Merge pull request #126 from mapzen/569-duplicate-points

Fix one cause of duplicate points.

rmarianski committed:
Add funicular to road kind rail

rmarianski committed:
Add transform to update parenthetical properties

zerebubuth committed:
Added ability to drop parenthetical features below some given zoom level.

rmarianski committed:
Merge pull request #127 from mapzen/add-funicular-roads

Add funicular to road kind rail

zerebubuth committed:
Merge pull request #128 from mapzen/update-parenthetical-properties

Filter on parenthetical properties

zerebubuth committed:
Add function to add construction state to stations.

zerebubuth committed:
Update function name - 'state' doesn't seem to be just about construction status.

zerebubuth committed:
Merge pull request #129 from mapzen/484-include-poi-state

Include station 'state' tag

zerebubuth committed:
Adjust tile rank score for stations to take into accound the different types of routes.

zerebubuth committed:
Remove temporary properties which shouldn't be public.

zerebubuth committed:
Add 'root_relation_id' for linking related features together in a site or public transport 'stop area' or 'stop area group'.

zerebubuth committed:
Merge pull request #130 from mapzen/506-adjust-tile-rank-for-stations

Adjust tile rank score for stations.

rmarianski committed:
Remove now unused landuse kind mapping transform

rmarianski committed:
Merge pull request #131 from mapzen/remove-unused-landuse-mapping-transform

Remove now unused landuse kind mapping transform

rmarianski committed:
Add uic_ref transform

rmarianski committed:
Merge pull request #132 from mapzen/add-uic_ref-transform

Add uic_ref transform

rmarianski committed:
Initial changelog for v0.9.0

rmarianski committed:
Merge pull request #133 from mapzen/changelog-v0.9.0

Initial changelog for v0.9.0

zerebubuth committed:
Shapely STRtree now uses the is_empty property to filter out empty geometries when building the tree.

zerebubuth committed:
Merge pull request #135 from mapzen/134-proxy-is-empty-property

Add `is_empty` property for geometry proxy.

rmarianski committed:
Add transform to convert admin_level to an int

rmarianski committed:
Merge pull request #136 from mapzen/transform-admin_level-int

Add transform to convert admin_level to an int

nvkelso committed:
connects to vector-datasource #680 (remove junk highway=minor and footpath)

zerebubuth committed:
Remove unused boundary transforms.

zerebubuth committed:
Merge pull request #138 from mapzen/687-yaml-transform-boundaries

Remove unused boundary transforms.

nvkelso committed:
Merge pull request #137 from mapzen/nvkelso/remove-junk-highway-types

connects to vector-datasource #680 (remove junk highway=minor and footpath)

zerebubuth committed:
Removed unused building kind calculation.

zerebubuth committed:
Merge pull request #139 from mapzen/646-use-yaml-for-buildings

Removed unused building kind calculation.

rmarianski committed:
kind=portage_way when whitewater=portage_way

rmarianski committed:
Delegate quantization to mapbox-vector-tile

rmarianski committed:
Remove the unused road_sort_key transform

rmarianski committed:
Merge pull request #140 from mapzen/roads-kind-portage_way

Update road kind for whitewater=portage_way

rmarianski committed:
Merge pull request #141 from mapzen/delegate-quantization-mvt

Delegate quantization to mapbox-vector-tile

rmarianski committed:
Merge pull request #142 from mapzen/remove-unused-road-sort_key-transform

Remove the unused road_sort_key transform

zerebubuth committed:
Fixed another source of duplicate points - not sure why this one appeared now?

zerebubuth committed:
Remove unused kind function and abandoned pistes filter.

zerebubuth committed:
Merge pull request #143 from mapzen/646-roads-yaml

Roads YAML cleanup and duplicate points fix.

zerebubuth committed:
Add function to convert any height present into meters.

zerebubuth committed:
Update conversion of lengths / heights to meters and add more units.

zerebubuth committed:
Merge pull request #145 from mapzen/677-add-waterfalls

Filter to convert heights into meters

nvkelso committed:
add transform for pois to convert capacity to int

nvkelso committed:
Merge pull request #146 from mapzen/nvkelso/pois-capacity-to-int

add transform for pois to convert capacity to int

zerebubuth committed:
Add function to convert elevations to meters.

zerebubuth committed:
Merge pull request #147 from mapzen/523-add-elevation-to-peaks

Add function to convert elevations to meters.

zerebubuth committed:
Sort POIs by elevation when they are peaks.

zerebubuth committed:
Extend peaks to include volcanos.

zerebubuth committed:
Merge pull request #148 from mapzen/524-peak-tile-kind-rank

Add sort order for peaks

zerebubuth committed:
Add an end_zoom parameter to control the behaviour of remove_duplicate_features.

zerebubuth committed:
Merge pull request #149 from mapzen/add-end-zoom-for-remove-duplicates

Add end zoom for remove duplicates.

rmarianski committed:
Add transform to add is_bicycle_route property

rmarianski committed:
Expand is_bicycle_route to include left/right/both

rmarianski committed:
Add transform to normalize bicycle left/right vals

rmarianski committed:
Normalize cycleway_both -> cycleway

rmarianski committed:
Don't label kind=rock|stone features

rmarianski committed:
Merge pull request #150 from mapzen/add-is_bicycle_route

Add transform to add is_bicycle_route property

rmarianski committed:
Merge pull request #151 from mapzen/restrict-label-generation

Don't label kind=rock|stone features

rmarianski committed:
Update bicycle related transform behavior

rmarianski committed:
Merge pull request #152 from mapzen/update-bicycle-related

Update bicycle related transform behavior

zerebubuth committed:
Make geometry types for label generation configurable.

zerebubuth committed:
Merge pull request #153 from mapzen/399-add-island-labels

Make geometry types for label generation configurable.

zerebubuth committed:
Don't convert lines or points to points for labelling - leave them as they were.

zerebubuth committed:
Merge pull request #154 from mapzen/399-add-island-labels

Don't convert line labels to points.

nvkelso committed:
v0.10 changelog

nvkelso committed:
Merge pull request #155 from mapzen/nvkelso/v0.10-changelog

v0.10 changelog

rmarianski committed:
Remove no longer used transform

rmarianski committed:
Merge pull request #156 from mapzen/normalize-ne-places

Remove no longer used transform

rmarianski committed:
Update building properties to use _ separator

The queries now return the building properties with _ as the separator.

rmarianski committed:
Merge pull request #157 from mapzen/building-height-level

Update building properties to use _ separator

rmarianski committed:
Update tags transform to convert wof l10n names

rmarianski committed:
Use pycountry to convert language codes

rmarianski committed:
Update behavior for route_name transform

rmarianski committed:
Merge pull request #159 from mapzen/route-name-transform

Update behavior for route_name transform

zerebubuth committed:
Start using true for boolean values rather than 'yes'.

zerebubuth committed:
Merge pull request #161 from mapzen/778-normalise-yes

Start using true for boolean values rather than 'yes'.

rmarianski committed:
Normalize l10n tags to use iso-639-3 code

rmarianski committed:
Removed pois transforms transitioned to yaml

rmarianski committed:
Remove unused road_kind transform

rmarianski committed:
Correct iso-639-2 lookup

rmarianski committed:
Support l10n lookups and country suffixes

rmarianski committed:
Merge pull request #162 from mapzen/transition-pois-transforms

Remove pois transforms transitioned to yaml

rmarianski committed:
Merge pull request #158 from mapzen/wof-name-l10n

Update tags transform to convert wof l10n names

rmarianski committed:
Add transform to drop properties with prefix

rmarianski committed:
Merge pull request #163 from mapzen/add-transform-drop-prefix

Add transform to drop properties with prefix

rmarianski committed:
Remove no longer necessary drop_keys

rmarianski committed:
Merge pull request #164 from mapzen/simplify-drop-keys

Remove no longer necessary drop_keys

rmarianski committed:
Add note that this repo is deprecated now

closes #166

ThomasG77 committed:
Fix ValueError: too many values to unpack when producing MVT tiles

rmarianski committed:
Merge pull request #168 from ThomasG77/fix-mvt

Fix "ValueError: too many values to unpack when producing MVT tiles"
